src/main/java/com/blamejared/crafttweaker/impl_native/event/entity/living/ExpandLivingSpawnEvent.java

```diff
@@ -0,0 +1,43 @@
+package com.blamejared.crafttweaker.impl_native.event.entity.living;
+
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import com.blamejared.crafttweaker_annotations.annotations.NativeTypeRegistration;
+import net.minecraft.world.World;
+import net.minecraftforge.event.entity.living.LivingSpawnEvent;
+import org.openzen.zencode.java.ZenCodeType;
+
+@ZenRegister
+@Document("vanilla/api/event/entity/living/MCLivingSpawnEvent")
+@NativeTypeRegistration(value = LivingSpawnEvent.class, zenCodeName = "crafttweaker.api.event.entity.living.MCLivingSpawnEvent")
+public class ExpandLivingSpawnEvent {
+    
+    @ZenCodeType.Getter("world")
+    public static World getWorld(LivingSpawnEvent internal) {
+        
+        if(internal.getWorld() instanceof World) {
+            return (World) internal.getWorld();
+        }
+        throw new IllegalStateException("the event is not fired on server or client world!");
```

Comment:
I feel this should silently exit and return null.

Looking at the code, it is always given a world to begin with (Always given entity.world which is `World`).

There also isn't a way for scripters to check if the world is instanceof World before running this method, so I feel this would lead to confusion.

Reply:
I has checked the event constructor usage. The field always is a World instance.

Reply:
Then you can leave it as is, just maybe clear it up like "IWorld instance was not an instance of World!"